.gitignore

```diff
@@ -1,32 +1,45 @@
-*~
-.DS_Store
+# Clojure / leiningen files
+pom.xml
+pom.xml.asc
+*.jar
+*.class
+/lib/
+/classes/
+/target/
+/checkouts/
 .lein-deps-sum
 .lein-repl-history
+.lein-plugins/
+.lein-failures
 .nrepl-port
-/pom.xml
-/pom.xml.asc
-/clojush*.jar
-classes/
-target/
+.cpcache/
+.lein-project-checksum
+
+# Editor and backup files
+*~
+.idea
+.gorilla-port
+.ipynb_checkpoints/
+.calva/
+.clj-kondo/
+.lsp
+
+# OS Files
+.DS_Store
+
+# Results and data directories
 results/
+/data/
+logs/
+
+# Unknown / Other
 .classpath
 .project
 .settings/ccw.repl.cmdhistory.prefs
-/data/
-logs/
-lib/
 /bin
-.lein-failures
 doc/
 cov/*
 out
-.idea
 clojush.iml
-.gorilla-port
 /clojush/
-.ipynb_checkpoints/
-.lein-project-checksum
 scratch.clj
```

Comment:
I also do something like this often, however I keep this rule in my local `.gitignore` and usually don't push it to the shared repository.

I don't think it is a big deal because the filename implies some kind of ephemeral playground that isn't of any purpose but it might be confusing to people reading through this file. What is a `scratch.clj`? Should I have one on my local clone of the repo?

Reply:
The "removals" were actually moved above.

I have no idea what scratch.clj is, but sounds like something that someone uses for scratch work.

Reply:
Yeah, scratch.clj would have been mine... I often want a file around for cutting/pasting purposes or trying things that I don't want to accidentally get left in a real file.